src/collective/cover/interfaces.py

```diff
@@ -30,14 +32,8 @@ def getSearchResults(self, filter_portal_types, rooted, document_base_url, searc
 
 
 class ICoverUIDsProvider(Interface):
-
-    def getUIDs(self):
-        """Get UUIDs associated with the object.
-            could be the UUID of the object or a
-            list of related UUIDs.
-
-        @return: iterable of UUIDs
-        """
+    """GenericUIDsProvider interface will be removed in collective.cover v1.7."""
+    warnings.warn(__doc__, DeprecationWarning)
```

Comment:
We still have some doubts about how this deprecation should work.

Usually, deprecation warnings still keep the functionality and, in the specified version in the message warning, is removed. The [`replace_with_objects` method in collective.cover list tile in version 1.1b1 was a good example.](https://github.com/collective/collective.cover/blob/1.1b1/src/collective/cover/tiles/list.py#L223)

The way it's now, the warning seems useless, because it'll be shown in an already broken package. For example, we have a package that uses collective.cover as a dependency that uses this class and these adapters, when trying to upgrade to collective.cover 1.5b1 we get:

```
uuids = ICoverUIDsProvider(obj).getUIDs()
TypeError: ('Could not adapt', <ATImage at /plone/my-image>, <InterfaceClass collective.cover.interfaces.ICoverUIDsProvider>)
```

https://travis-ci.org/plonegovbr/brasil.gov.tiles/jobs/245449596#L1565

Any idea how we should solve this problem? Shouldn't the removal of these classes [(and adapters)](https://github.com/collective/collective.cover/pull/714/files#diff-2b38db85530a6440d8600df43b0f963bL88) be reverted (we can do a PR, we're not so sure exactly where to put the `warnings`)? If I add them "by hand" in a hacky manner in my eggs folder, the errors disappear.